code/api/views.py

```diff
@@ -325,6 +325,29 @@ def post(self, request, author_id):
                     }
                 )
 
+                if not post_created:
+                    categories_to_remove = [category_obj.category 
+                        for category_obj in received_post.categories.all()]
+                else:
+                    categories_to_remove = []
```

Comment:
Don't think this matters much, but do we really need to check update categories here? I don't think posts will ever be updated via the inbox. Should be fine if you want to keep it like this, but we should only ever need to set the categories the first time the post is received (I think).

Reply:
It's probably less error-prone to just do a simple create post like you do in the post() function in socialDistribution/views.py. If you do want to do fancy logic to update categories, it would probably be best if you could move this code to a method somewhere,  then call that method everywhere this logic is needed (Note that this code also has the delete bug. Putting all this logic in a single method means we only need to fix that bug once.)

Reply:
Hmm, I think I got confused by the get_or_create no line 310. This functionality should be moved to the post endpoint then. I'll make an issue for that and assign myself to it.

Reply:
Oh, I see. The reason I did get_or_create was to handle the case where multiple users on our server are sent the same remote post, since right now we only save one InboxPost for all the authors that it is sent to. This isn't necessarily the best design though, so we could change it down the line.

Reply:
I see, that makes sense. I will make an issue for implementing the post endpoint and assign myself to it and while dealing with this issue, I'll move the category functionality to that endpoint. One more thing was wondering about. This might be a bit off topic for this pr, but for some endpoints, we'll be having logic duplication right? For example, editing a post and creating a post in api/views.py will have the same logic as in socialDistribution/views.py. I was wondering if we should move those functionalities to a method somewhere as you suggested for categories?

Reply:
Ya, it would be ideal to move what we can to methods so that we are only writing the code once. It's just a bit late for some areas of the code base as refactoring now is kinda time-consuming and we are rushed. It would be best to keep logic duplication as low as possible since it can lead to bugs and make it more difficult to add more features. 

If you are working on a feature that could use refactoring, please go ahead and do it. It could save us from a nasty time-consuming bug in the future. I think the most important thing is that we just do our best to avoid duplicating logic with any new code we write.

Reply:
Ok, that sounds good! 👌 I'll be mindful of that
Thanks a lot for the feedback! ♥